packages/arb-token-bridge-ui/src/util/NumberUtils.ts

```diff
@@ -9,23 +9,89 @@ export function formatUSD(value: number) {
   return `$${formattedValue}`
 }
 
-export function formatNumber(value: number, maximumFractionDigits: number = 6) {
-  return value.toLocaleString(undefined, {
+export enum Decimals {
+  None = 0,
+  Short = 1,
+  Compact = 3,
+  Standard = 4,
+  Long = 5,
+  Token = 18
+}
+
+export const formatNumber = <T extends number | BigNumber>(
+  number: T,
+  options: {
+    decimals?: T extends number ? never : number
+    maximumFractionDigits?: number
+    notation?: 'compact' | 'standard'
+  } = {
+    maximumFractionDigits: 3,
+    notation: 'standard'
+  }
+): string => {
+  const { decimals, notation, maximumFractionDigits } = options
+  const value: number = BigNumber.isBigNumber(number)
+    ? parseFloat(utils.formatUnits(number, decimals ?? Decimals.Token))
+    : number
+
+  return Intl.NumberFormat('en', {
+    notation,
     maximumFractionDigits
-  })
+  }).format(value)
 }
 
-export function formatBigNumber(
-  value: BigNumber,
-  decimals: number = 18,
-  maximumFractionDigits?: number
-) {
-  if (value.isZero()) {
+export const formatAmount = <T extends number | BigNumber>(
```

Comment:
Can we add some multiline comments above both of the methods to make it clear when to use which method?
Their names are too similar to immediately understand which should be used in which case

Reply:
Make sense, I'll add a comment

Reply:
We made it simpler: you only want to use `formatAmount`, `formatNumber` is not exported anymore